martian-filetypes/src/gzip_file.rs

```diff
@@ -67,17 +66,21 @@ where
 {
     fn read(&self) -> Result<T, Error> {
         let decoder = GzDecoder::new(self.buf_reader()?);
-        Ok(<Self as FileTypeIO<T>>::read_from(decoder)
-            .with_context(|e| ErrorContext::ReadContext(self.as_ref().into(), e.to_string()))?)
+        Ok(<Self as FileTypeIO<T>>::read_from(decoder).map_err(|e| {
+            let context = ErrorContext::ReadContext(self.as_ref().into(), e.to_string());
+            e.context(context)
+        })?)
```

Comment:
I used `.map_err` here to keep the error message as close as possible to the original; `anyhow`s `.with_context` doesn't have access to the original error, so it is not a good substitute to `failure`s `.with_context`.